crypto/cms/cms_smime.c

```diff
@@ -307,26 +313,37 @@ int CMS_verify(CMS_ContentInfo *cms, STACK_OF(X509) *certs,
 
     /* Attempt to verify all signers certs */
 
-    if (!(flags & CMS_NO_SIGNER_CERT_VERIFY)) {
+    if (!(flags & CMS_NO_SIGNER_CERT_VERIFY) || (flags & CMS_CADES)) {
+        if (flags & CMS_CADES)
+            si_chains = OPENSSL_zalloc(scount * sizeof(si_chains[0]));
```

Comment:
probably need to check for alloc failure here. At the moment it just ignores the chain